susemanager-utils/susemanager-sls/src/modules/reboot_info.py

```diff
@@ -51,7 +51,8 @@ def reboot_required():
     elif __grains__["os_family"] == "RedHat":
         cmd = (
             "dnf -q needs-restarting -r"
-            if __grains__["osmajorrelease"] >= 8
+            # In RHEL6 and clones 'osmajorrelease' is a string
+            if int(__grains__["osmajorrelease"]) >= 8
             else "needs-restarting -r"
```

Comment:
Is `needs-restarting -r` always available on RHEL 6 or can this cause another error?

Reply:
Just for clarifying here, this changes are not meant to bring the "reboot info" feature to the unsupported RHEL6 clients, but at least to avoid making the function to crash, which leads to failed actions.

So, `needs-restarting` is provided by `yum-utils` package and available and installed in RHEL 6 registered clients.

OTOH the `-r` argument is not supported in RHEL 6, leading to this error:

```
[root@6dbee07d70e3 /]# rpm -qf /usr/bin/needs-restarting 
yum-utils-1.1.30-40.el6.noarch
[root@6dbee07d70e3 /]# needs-restarting -r
Usage: 
    needs-restarting: Report a list of process ids of programs that started 
                    running before they or some component they use were updated.
    

needs-restarting: error: no such option: -r
[root@6dbee07d70e3 /]# echo $?
2
```

The logic in `reboot_info` module just tries to check if retcode from `needs-restarting` is equal to `1`, which won't happen for RHEL 6 (always retcode 2 when passing `-r`).

I wanted to avoid adding a particular condition for the case of unsupported RHEL 6.